code/modules/atmospherics/machinery/components/unary_devices/cryo.dm

```diff
@@ -146,20 +138,30 @@
 		icon_state = "pod-off"
 		add_overlay("cover-off")
 
-/obj/machinery/atmospherics/components/unary/cryo_cell/proc/run_anim(anim_up, image/occupant_overlay)
+/obj/machinery/atmospherics/components/unary/cryo_cell/proc/run_anim(is32x32, image/occupant_overlay, anim_up = TRUE)
 	if(!on || !occupant || !is_operational())
 		running_anim = FALSE
 		return
 	cut_overlays()
-	if(occupant_overlay.pixel_y != 23) // Same effect as occupant_overlay.pixel_y == 22 || occupant_overlay.pixel_y == 24
-		anim_up = occupant_overlay.pixel_y == 22 // Same effect as if(occupant_overlay.pixel_y == 22) anim_up = TRUE ; if(occupant_overlay.pixel_y == 24) anim_up = FALSE
+	if(overlay_pixel_y != 1) // Same effect as overlay_pixel_y == 0 || occupant_overlay.pixel_y == 2
+		anim_up = overlay_pixel_y == 0 // Same effect as if(overlay_pixel_y == 0) anim_up = TRUE ; if(overlay_pixel_y == 2) anim_up = FALSE
 	if(anim_up)
-		occupant_overlay.pixel_y++
+		overlay_pixel_y++
 	else
-		occupant_overlay.pixel_y--
-	add_overlay(occupant_overlay)
+		overlay_pixel_y--
+	add_masked_overlay(is32x32, occupant_overlay)

```

Comment:
Can you use looped animate here instead of reapplying ?

Reply:
`animate` doesn't work for overlays

Reply:
You can use `vis_contents` (technically also images but that's not the best use case here)

Reply:
It is impossible to use blend mode with `vis_contents`, so it wouldn't work for non-32x32.